src/addresses/address-details/AddressDetails.component.test.tsx

```diff
@@ -0,0 +1,129 @@
+      </AuthContext.Provider>,
+    );
+
+    expect(screen.getByText('Address')).toBeInTheDocument();
+    expect(screen.getByText('AddressSize')).toBeInTheDocument();
+    expect(screen.getByText('AutoCreated')).toBeInTheDocument();
+    expect(screen.getByText('MessageCount')).toBeInTheDocument();
+    expect(screen.getByText('RoutingTypes')).toBeInTheDocument();
+  });
+
+  it('should renders loading spinner initially when data not available', () => {
+    (useJolokiaServiceReadAddressAttributes as jest.Mock).mockReturnValueOnce({
+      data: null,
+      isSuccess: false,
+      error: null,
+    });
+
+    render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+
+    expect(screen.getByText('attributes')).toBeInTheDocument();
+  });
+
+  it('should filters the data based on search input', () => {
+    render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+
+    const searchInput = screen.getByPlaceholderText('search');
+    fireEvent.change(searchInput, { target: { value: 'Message' } });
+
+    expect(screen.getByText('MessageCount')).toBeInTheDocument();
+    expect(screen.queryByText('Address')).not.toBeInTheDocument();
+  });
+
+  it('should clear the search input and reset the initial data when search input is cleared', () => {
+    render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+    //initial data rendered
+    expect(screen.getByText('Address')).toBeInTheDocument();
+    expect(screen.getByText('AddressSize')).toBeInTheDocument();
+    expect(screen.getByText('AutoCreated')).toBeInTheDocument();
+    expect(screen.getByText('MessageCount')).toBeInTheDocument();
+    expect(screen.getByText('RoutingTypes')).toBeInTheDocument();
+
+    const searchInput = screen.getByPlaceholderText('search');
+    fireEvent.change(searchInput, { target: { value: 'Message' } });
+
+    fireEvent.change(searchInput, { target: { value: '' } });
+    expect(searchInput).toBeInTheDocument();
+    //Verify that the initial data is displayed again
+    expect(screen.getByText('Address')).toBeInTheDocument();
+    expect(screen.getByText('AddressSize')).toBeInTheDocument();
+    expect(screen.getByText('AutoCreated')).toBeInTheDocument();
+    expect(screen.getByText('MessageCount')).toBeInTheDocument();
+    expect(screen.getByText('RoutingTypes')).toBeInTheDocument();
+  });
+
+  it('should update the current page when pagination controls are used', () => {
+    const { container } = render(
+      <AuthContext.Provider value="mock-token">
+        <AddressDetails name="DLQ" />
+      </AuthContext.Provider>,
+    );
+    const currentPage = container.querySelector(
+      '#pagination-options-menu-top-top-pagination > nav > div:nth-child(3) > input',
+    );
+    expect(currentPage).toHaveValue(1);
+  });
```

Comment:
Can you update this test so that it clicks on the next page and asserts that the page number is 2 please?